File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string sample);
int count_words(string sample);
int count_sentences(string sample);

int main(void)
{
    string text = get_string("Text: ");

//    printf("%i letters\n", count_letters(text));
//    printf("%i words\n", count_words(text));
//    printf("%i sentences\n", count_sentences(text));

    //Calculates index
    float L = (float) count_letters(text) / (float) count_words(text) * 100;
    float S = (float) count_sentences(text) / (float) count_words(text) * 100;
    int index = round((0.0588 * L) - (0.296 * S) - 15.8);
//    printf("L = %.2f and S = %.2f\n", L, S);

    // Prints result
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}




// Counts the letters in the text
int count_letters(string sample)
{
    int letters = 0;
    for (int i = 0, n = strlen(sample); i < n; i++)
    {
        if (isalpha(sample[i]))
        {
            letters++;
        }
    }
    return letters;
}


// Counts the words in the text
int count_words(string sample)
{
    int words = 1;
    for (int j = 0, n = strlen(sample); j < n; j++)
    {
        if (isspace(sample[j]))
        {
            words++;
        }
    }
    return words;
}


// Counts the sentences in the text
int count_sentences(string sample)
{
    int sentences = 0;
    for (int k = 0, n = strlen(sample); k < n; k++)
    {
        if (sample[k] == '.' || sample[k] == '!' || sample[k] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: week06/pset6/sentimental-readability/readability.py
# TODO

from cs50 import get_string


# Main body
def main():

    # Get user input
    text = get_string("Text: ")

    # Calculate index
    L = count_letters(text) / count_words(text) * 100
    S = count_sentences(text) / count_words(text) * 100
    index = round((0.0588 * L) - (0.296 * S) - 15.8)

    # Print result
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


# Count letters
def count_letters(sample):
    letters = 0
    for i in range(len(sample)):
        if sample[i].isalpha():
            letters += 1
    return letters


# Count words
def count_words(sample):
    words = 1
    for i in range(len(sample)):
        if sample[i].isspace():
            words += 1
    return words


# Count sentences
def count_sentences(sample):
    sentences = 0
    for i in range(len(sample)):
        if sample[i] == "." or sample[i] == "!" or sample[i] == "?":
            sentences += 1
    return sentences


main()